tests/components/octoprint/test_button.py

```diff
@@ -192,3 +192,78 @@ async def test_stop_job(hass: HomeAssistant) -> None:
         )
 
         assert len(stop_command.mock_calls) == 0
+
+
+async def test_shutdown_system(hass: HomeAssistant) -> None:
+    """Test the shutdown system button."""
+    await init_integration(hass, BUTTON_DOMAIN)
+
+    coordinator: OctoprintDataUpdateCoordinator = hass.data[DOMAIN]["uuid"][
```

Comment:
You cannot access hass.data in tests

Reply:
Why can't I? I have now removed it because it's not needed in my tests but it was working.
I've copied it from the other existing tests in this file. So are they wrong too?
What would be the correct way (e.g. for the other tests)?

Reply:
We don't want the tests to directly mess with integration internals, in this case you're testing the button platform and you should need access to the data coordinator to do that. Instead, your test should, as a general rule of thumb, check the state in the state machine, call the button's press service, and by mocking the 3rd party supporting library check that pressing the button has the intended side effects.